File: AtlasWeb/Core/wwwroot/app/System/Models/UserResourceSelectionModel.ts
import { BaseModel } from 'Fw/Models/BaseModel';
import { ResSection } from 'NebulaClient/Model/AtlasClientModel';
import { Resource, ExaminationQueueDefinition } from 'NebulaClient/Model/AtlasClientModel';

export class UserResourceSelectionModel extends BaseModel {
    public OutPatientResourceList: Array<Resource>;
    public InPatientResourceList: Array<Resource>;
    public SecMasterResourceList: Array<Resource>;
    public QueueList: Array<ExaminationQueueDefinition>;
    public SelectedOutPatientResource: ResSection;
    public SelectedInPatientResource: ResSection;
    public SelectedSecMasterResource: ResSection;
    public SelectedQueue: ExaminationQueueDefinition;

    constructor() {
        super();
        this.DefaultButtonsVisible = false;
    }
}

File: AtlasWeb/Core/Models/UserResourceSelectionModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using TTObjectClasses;

namespace Core.Models
{
    public class UserResourceSelectionModel
    {
        public List<Resource> OutPatientResourceList { get; set; }
        public List<Resource> InPatientResourceList { get; set; }
        public List<Resource> SecMasterResourceList { get; set; }
        public List<ExaminationQueueDefinition> QueueList { get; set; }
        public ResSection SelectedOutPatientResource { get; set; }
        public ResSection SelectedInPatientResource { get; set; }
        public ResSection SelectedSecMasterResource { get; set; }
        public ExaminationQueueDefinition SelectedQueue { get; set; }
    }
}
